trie/node.go

```diff
@@ -45,6 +45,23 @@ type (
 	}
 	hashNode  []byte
 	valueNode []byte
+
+	//fullnodeEncoder is a type used exclusively for encoding. Briefly instantiating
+	// a fullnodeEncoder and initializing with existing slices is less memory
+	// intense than using the fullNode type.
+	fullnodeEncoder struct {
+		Children [17][]byte
+		flags    nodeFlag
+	}
+
+	//shortNodeEncoder is a type used exclusively for encoding. Briefly instantiating
+	// a shortNodeEncoder and initializing with existing slices is less memory
+	// intense than using the shortNode type.
+	shortNodeEncoder struct {
```

Comment:
qq, does `shortNodeEncoder` need to implement the node interface?

It's just an encoder which coverts the node object to byte format?

Reply:
just have a try, it's technically possible

Reply:
Good thinking there!

![Good thinking](https://www.memesportal.com/wp-content/uploads/2021/04/2pzj16-1024x614.jpg)